lib/Sema/CSSimplify.cpp

```diff
@@ -3825,8 +3826,41 @@ ConstraintSystem::SolutionKind ConstraintSystem::simplifyMemberConstraint(
 
     // FIXME(diagnostics): If there were no viable results, but there are
     // unviable ones, we'd have to introduce fix for each specific problem.
-    if (!result.UnviableCandidates.empty())
+    if (!result.UnviableCandidates.empty()) {
+      // Check if we have unviable candidates whose reason for rejection
+      // is either UR_InstanceMemberOnType or UR_TypeMemberOnInstance
+      auto unviable = [&](const std::pair<OverloadChoice,
+                          MemberLookupResult::UnviableReason> &e) {
+        return e.second == MemberLookupResult::UR_InstanceMemberOnType ||
+               e.second == MemberLookupResult::UR_TypeMemberOnInstance;
+      };
+      
+      // If we do, then allow them
+      if (llvm::all_of(result.UnviableCandidates, unviable)) {
+        
+        auto *fix = AllowTypeOrInstanceMember::create(*this, baseTy, member,
+                                                      getConstraintLocator(locator));
+        
+        if (recordFix(fix)) {
+          // We couldn't record a fix, so bail out
+          return SolutionKind::Error;
+        }
+        
+        // The fix was successful, so let's add it to the overload set
+        // and return the solution as solved
+        auto overloadChoices = SmallVector<OverloadChoice, 4>();
+        
+        for (auto &pair : result.UnviableCandidates) {
+          overloadChoices.push_back(pair.first);
```

Comment:
Should `OverloadChoice` be moved to new vector?

Reply:
Yes as I can’t pass `UnviableCandidates` directly as it’s an array of `std::pair<...>`, so I need to create a vector to store the overload choices, unless there’s another way to extract them directly?

Reply:
Sorry, you can keep the array, what I mean is to use `std::move` to move elements over from unviable list, instead of copying :)